### PR TITLE
config/settings: Add microshift preset to helper

### PR DESCRIPTION
**Pull request by praveenkumar**
**Fixes:** Issue #3576

### CODE REVIEW

File: pkg/crc/config/settings.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func RegisterSettings(cfg *Config) {
 
 	// Preset setting should be on top because CPUs/Memory config depend on it.
 	cfg.AddSetting(Preset, string(preset.OpenShift), validatePreset, RequiresDeleteAndSetupMsg,
-		fmt.Sprintf("Virtual machine preset (valid values are: %s, %s and %s)", preset.Podman, preset.OpenShift, preset.OKD))
+		fmt.Sprintf("Virtual machine preset (valid values are: %s, %s, %s and %s)", preset.Podman, preset.OpenShift, preset.Microshift, preset.OKD))
```
Comment (cfergeau) [inline pkg/crc/config/settings.go]: A more future-proof alternative is to add `func preset.AllPresets() []preset.Preset` and use this in `RegisterSettings` and `validatePreset`

File: pkg/crc/preset/preset.go [hunk 2]
```diff
@@ -15,16 +15,20 @@ const (
...
-		return string(OpenShift)
-	case OKD:
-		return string(OKD)
-	case Microshift:
-		return string(Microshift)
+	for pSet, pString := range presetMap() {
+		if pSet == preset {
+			return pString
+		}
 	}
```
Comment (cfergeau) [inline pkg/crc/preset/preset.go:37]: you have a map, you don't need to iterate over it
```
presetStr, presetExists := presetMap()[preset]
if !presetExists {
    return "invalid"
}
return presetStr
```

File: pkg/crc/preset/preset.go [hunk 3]
```diff
@@ -15,16 +15,20 @@ const (
...
 )
 
+func presetMap() map[Preset]string {
+	return map[Preset]string{
+		Podman:     string(Podman),
+		OpenShift:  string(OpenShift),
+		OKD:        string(OKD),
+		Microshift: string(Microshift),
+	}
+}
```
Comment (cfergeau) [inline pkg/crc/preset/preset.go:31]:
```
var presetMap = map[Preset]string {
...
}
```
should be enough, and allow us to create the map only once?

File: pkg/crc/config/settings.go [hunk 4]
```diff
@@ -70,7 +70,7 @@ func RegisterSettings(cfg *Config) {
 
 	// Preset setting should be on top because CPUs/Memory config depend on it.
 	cfg.AddSetting(Preset, string(preset.OpenShift), validatePreset, RequiresDeleteAndSetupMsg,
-		fmt.Sprintf("Virtual machine preset (valid values are: %s, %s and %s)", preset.Podman, preset.OpenShift, preset.OKD))
+		fmt.Sprintf("Virtual machine preset (valid values are: %s)", preset.AllPresets()))
```
Comment (cfergeau) [inline pkg/crc/config/settings.go:73]: What does the string looks like ? I don't know how go converts an array to a string.

Comment (praveenkumar) [reply pkg/crc/config/settings.go:73]: @cfergeau it have `Virtual machine preset (valid values are: [openshift okd podman microshift])` if we really want to have as string then another helper is required.

Comment (cfergeau) [reply pkg/crc/config/settings.go:73]: `strings.Join(preset.AllPresets, ", ")` might work.

Comment (praveenkumar) [reply pkg/crc/config/settings.go:73]: > strings.Join(preset.AllPresets, ", ") might work

No, beacuse it complain `preset.AllPresets` is not `[]string` type so not able to use.

Comment (cfergeau) [reply pkg/crc/config/settings.go:73]: Let's say `[openshift okd podman microshift]` is good enough if you don't want the extra code.

File: pkg/crc/preset/preset.go (hunk 3 above)
Comment (praveenkumar) [reply pkg/crc/preset/preset.go:31]: updated.

File: pkg/crc/preset/preset.go (hunk 2 above)
Comment (praveenkumar) [reply pkg/crc/preset/preset.go:37]: updated

**Review by cfergeau — APPROVED** [review]

**Pull request merged**